cloudinit/net/__init__.py

```diff
@@ -1004,7 +1004,7 @@ def get_interfaces_by_mac_on_linux(blacklist_drivers=None) -> dict:
 
     Bridges and any devices that have a 'stolen' mac are excluded."""
     ret: dict = {}
-    driver_map: dict = {}
+
```

Comment:
I think we can remove the raise_duplicate_mac_error variable and its check now that the hv_netvsc processing bit is refactored out.
if mac in ret:
    if driver in in ("fsl_enetc", "mscc_felix", "qmi_wwan"):
         ...
         continue
     raise RuntimeError(...)